src/main/java/seedu/address/logic/commands/CommandUtil.java

```diff
@@ -0,0 +1,87 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.module.Code;
+import seedu.address.model.module.Module;
+import seedu.address.model.module.Semester;
+import seedu.address.model.module.Year;
+
+//@@author alexkmj
+/**
+ * Contains utility methods used for executing commands in the various
+ * Command classes.
+ */
+public class CommandUtil {
+    public static final String MESSAGE_NO_SUCH_MODULE = "No such module.";
+    public static final String MESSAGE_MULTIPLE_MODULE_ENTRIES = "There exists"
+            + " more than one module that matches the target code, and target"
+            + " year and target semester is not specified.";
+
+    /**
+     * Returns the targeted module.
+     * <p>
+     * Checks if module specified by {@code targetCode} exist. If multiple
+     * module entries matches {@code targetCode}, check if {@code targetYear}
+     * and {@code targetSemester} has been specified. If all check passes, the
+     * targeted module is returned.
+     *
+     * @param model model containing the transcript
+     * @return targeted module
+     * @throws CommandException thrown when specified module does not exist or
+     * there are multiple module entries matching the {@code targetCode} but
+     * {@code targetYear} or {@code targetSemester} was not specified
+     */
+    public static Module getUniqueTargetModule(Model model, Code targetCode,
+            Year targetYear, Semester targetSemester) throws CommandException {
+        requireNonNull(targetCode);
+
+        // Returns the number of modules with target code, year, and semester.
+        List<Module> filteredModule = model.getFilteredModuleList()
```

Comment:
@jeremiah-ang  I actually thought about it before. But in doing so, I would have to iterate through the list of modules thrice:
1. `hasModule`
2. `hasMultipleInstance`
3. `findModule`
Instead of:
1. getting the filtered list of modules
And then check the size of filtered list; 0, 1, or more in it before deleting.
What do you think

Reply:
Then I think can improve the findModule to throw multipleInstanceException instead. If not we keep repeating this block of code. Otherwise it's still O(n) anyway and n <= 100 so it's not a big issue.